File: models/position.go
package models

import "github.com/shopspring/decimal"

// Position ...
type Position struct {
	BaseModel
	FundID            string          `json:"fund_id"`
	Fund              Fund            `json:"-"`
	AccountID         uint64          `json:"account_id"`
	Account           Account         `json:"-"`
	OpenAveragePrice  decimal.Decimal `json:"open_average_price" sql:"DECIMAL(32,16)"`
	CloseAveragePrice decimal.Decimal `json:"close_average_price" sql:"DECIMAL(32,16)"`
	LiquidationPrice  decimal.Decimal `json:"liquidation_price" sql:"DECIMAL(32,16)"`
	OpenType          string          `json:"open_type"`
	Side              string          `json:"side"` // sell or buy
	State             uint            `gorm:"default:0" json:"state"`
	OpenVolume        uint64          `json:"open_volume"`
	CloseVolume       uint64          `json:"close_volume"`
}

File: unicorn/app/models/position.rb
# frozen_string_literal: true

class Position < ApplicationRecord
end

# == Schema Information
#
# Table name: positions
#
#  id                                         :bigint           not null, primary key
#  close_average_price(平仓均价)              :decimal(32, 16)  default(0.0)
#  close_volume(已平仓位)                     :bigint           default(0), not null
#  liquidation_price(强平价格)                :decimal(32, 16)  default(0.0)
#  open_average_price(开仓均价)               :decimal(32, 16)  default(0.0)
#  open_type(开仓方式 全仓cross 逐仓isolated) :string(8)        not null
#  open_volume(开仓量)                        :bigint           not null
#  side(sell or buy)                          :string(8)        not null
#  state(状态)                                :integer          default(0), not null
#  created_at                                 :datetime         not null
#  updated_at                                 :datetime         not null
#  account_id(账户)                           :bigint           not null
#  instrument_id(产品)                        :string(32)       not null
#
